### ISSUE TITLE
7.0.0.dev Upstream "Proxy Authentication Required", possibly related to #4355

### ISSUE THREAD

**Issue by sickcodes** [mitmproxy/mitmproxy#4356]
#### Problem Description
407 Authentication required
Setting an upstream proxy following options in all of mitmproxy, mitmweb and inside python scripts :

#### System Information
Mitmproxy: 7.0.0.dev
Python:    3.9.1
OpenSSL:   OpenSSL 1.1.1i

#### Steps to reproduce the behavior:

`mitmweb --mode 'upstream:http://localhost:1337' --upstream-auth testuser:testpass`

`mitmproxy --mode 'upstream:http://localhost:1337' --upstream-auth testuser:testpass`

```python
options = Options(
    listen_host='127.0.0.1',
    listen_port=8080,
    upstream_auth='testuser:testpass',
    mode='upstream:http://localhost:1337',
)
```

Then set mitmproxy, mitmweb or set upstream auth in Python.

I tested on multiple auth required proxies, can test easily with tinyproxy

```bash
mkdir -p ./tinyproxy

PROXY_USER=testuser
PROXY_PASS=testpass
PROXY_PORT=1337

tee ./tinyproxy/tinyproxy.conf <<EOF
User tinyproxy
Group tinyproxy
Listen 0.0.0.0
Port ${PROXY_PORT}
Timeout 600
LogFile "./tinyproxy/tinyproxy.log"
LogLevel Info
MaxRequestsPerChild 0
MaxClients 100
StartServers 10
# Allow 0.0.0.0
# Allow ::1
BasicAuth ${PROXY_USER} ${PROXY_PASS}
DisableViaHeader Yes
EOF

tinyproxy -d -c ./tinyproxy/tinyproxy.conf

#curl -x testuser:testpass@localhost:1337 api.ipify.org
```

## Error

```python
Addon error: Traceback (most recent call last):
  File "./lib/python3.9/site-packages/mitmproxy/addons/view.py", line 568, in error
    self.update([f])
  File "./lib/python3.9/site-packages/mitmproxy/addons/view.py", line 612, in update
    self.sig_view_update.send(self, flow=f)
  File "./lib/python3.9/site-packages/blinker/base.py", line 266, in send
    return [(receiver, receiver(sender, **kwargs))
  File "./lib/python3.9/site-packages/blinker/base.py", line 266, in <listcomp>
    return [(receiver, receiver(sender, **kwargs))
  File "./lib/python3.9/site-packages/mitmproxy/tools/web/master.py", line 57, in _sig_view_update
    app.ClientConnection.broadcast(
  File "./lib/python3.9/site-packages/mitmproxy/tools/web/app.py", line 214, in broadcast
    message = json.dumps(kwargs, ensure_ascii=False).encode("utf8", "surrogateescape")
  File "/usr/lib/python3.9/json/__init__.py", line 234, in dumps
    return cls(
  File "/usr/lib/python3.9/json/encoder.py", line 199, in encode
    chunks = self.iterencode(o, _one_shot=True)
  File "/usr/lib/python3.9/json/encoder.py", line 257, in iterencode
    return _iterencode(o, 0)
  File "/usr/lib/python3.9/json/encoder.py", line 179, in default
    raise TypeError(f'Object of type {o.__class__.__name__} '
TypeError: Object of type bytes is not JSON serializable

Addon error: Traceback (most recent call last):
  File "./lib/python3.9/site-packages/mitmproxy/addons/view.py", line 612, in update
    self.sig_view_update.send(self, flow=f)
  File "./lib/python3.9/site-packages/blinker/base.py", line 266, in send
    return [(receiver, receiver(sender, **kwargs))
  File "./lib/python3.9/site-packages/blinker/base.py", line 266, in <listcomp>
    return [(receiver, receiver(sender, **kwargs))
  File "./lib/python3.9/site-packages/mitmproxy/tools/web/master.py", line 57, in _sig_view_update
    app.ClientConnection.broadcast(
  File "./lib/python3.9/site-packages/mitmproxy/tools/web/app.py", line 214, in broadcast
    message = json.dumps(kwargs, ensure_ascii=False).encode("utf8", "surrogateescape")
  File "/usr/lib/python3.9/json/__init__.py", line 234, in dumps
    return cls(
  File "/usr/lib/python3.9/json/encoder.py", line 199, in encode
    chunks = self.iterencode(o, _one_shot=True)
  File "/usr/lib/python3.9/json/encoder.py", line 257, in iterencode
    return _iterencode(o, 0)
  File "/usr/lib/python3.9/json/encoder.py", line 179, in default
    raise TypeError(f'Object of type {o.__class__.__name__} '
TypeError: Object of type bytes is not JSON serializable
```

Request then ultimately fails and returns `407 Proxy Authentication Required`

**Comment by mhils** [comment]
The upstream_auth option is not implemented yet in the new core, although it should be relatively trivial to add. The error your are seeing is unrelated and will be fixed by #4357.


I'll close this here in favor of  https://github.com/mitmproxy/mitmproxy/issues/4348. Thanks!

**Comment by sickcodes** [comment]
Fantastic, fixed in https://github.com/mitmproxy/mitmproxy/pull/4635
